test/common/lib/platform_user/platform_user.c

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+#include "platform.h"
+
+uint64_t CxPlatPerfFreq;
+CXPLAT_PROCESSOR_INFO* CxPlatProcessorInfo;
+CXPLAT_PROCESSOR_GROUP_INFO* CxPlatProcessorGroupInfo;
+uint32_t CxPlatProcessorCount;
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+_Must_inspect_result_
+XDP_STATUS
+CxPlatGetProcessorGroupInfo(
+    _In_ LOGICAL_PROCESSOR_RELATIONSHIP Relationship,
+    _Outptr_ _At_(*Buffer, __drv_allocatesMem(Mem)) _Pre_defensive_
+        PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX* Buffer,
+    _Out_ PDWORD BufferLength
+    );
+
+XDP_STATUS
+CxPlatProcessorInfoInit(
+    void
+    )
+{
+    XDP_STATUS Status = XDP_STATUS_SUCCESS;
+    DWORD InfoLength = 0;
+    SYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX* Info = NULL;
+    uint32_t ActiveProcessorCount = 0, MaxProcessorCount = 0;
+    Status =
+        CxPlatGetProcessorGroupInfo(
+            RelationGroup,
+            &Info,
+            &InfoLength);
+    if (XDP_FAILED(Status)) {
+        goto Error;
+    }
+
+    CXPLAT_DBG_ASSERT(InfoLength != 0);
+    CXPLAT_DBG_ASSERT(Info->Relationship == RelationGroup);
+    CXPLAT_DBG_ASSERT(Info->Group.ActiveGroupCount != 0);
+    CXPLAT_DBG_ASSERT(Info->Group.ActiveGroupCount <= Info->Group.MaximumGroupCount);
+    if (Info->Group.ActiveGroupCount == 0) {
+        Status = XDP_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    for (WORD i = 0; i < Info->Group.ActiveGroupCount; ++i) {
+        ActiveProcessorCount += Info->Group.GroupInfo[i].ActiveProcessorCount;
+        MaxProcessorCount += Info->Group.GroupInfo[i].MaximumProcessorCount;
+    }
+
+    CXPLAT_DBG_ASSERT(ActiveProcessorCount > 0);
+    CXPLAT_DBG_ASSERT(ActiveProcessorCount <= UINT16_MAX);
+    if (ActiveProcessorCount == 0 || ActiveProcessorCount > UINT16_MAX) {
+        Status = XDP_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    CXPLAT_FRE_ASSERT(CxPlatProcessorInfo == NULL);
+    CxPlatProcessorInfo =
+        CXPLAT_ALLOC_NONPAGED(
+            ActiveProcessorCount * sizeof(CXPLAT_PROCESSOR_INFO),
+            QUIC_POOL_PLATFORM_PROC);
```

Comment:
All QUIC* code must be CXPLAT*

Reply:
Interesting, there is no QUIC_POOL_PLATFORM_PROC defined in this repo